it/v2/connectors/PersistenceConnectorSpec.scala

```diff
@@ -921,9 +921,108 @@ class PersistenceConnectorSpec
     }
   }
 
-  private def generateDepartureResponse(departureId: MovementId) =
-    DepartureResponse(
-      _id = departureId,
+  "GET /traders/:EORI/movements/arrivals" - {
+
+    implicit val hc = HeaderCarrier()
+    val eori        = arbitrary[EORINumber].sample.get
+
+    def targetUrl(eoriNumber: EORINumber) = s"/transit-movements/traders/${eoriNumber.value}/movements/arrivals"
+
+    "on success, return a list of arrival IDs" in {
+      lazy val arrivalIdList = Gen.listOfN(3, arbitrary[MovementId]).sample.get
+      lazy val messageResponse = arrivalIdList.map(
+        id => generateMovementResponse(id)
+      )
+
+      server.stubFor(
+        get(
+          urlEqualTo(targetUrl(eori))
+        )
+          .willReturn(
+            aResponse()
+              .withStatus(OK)
+              .withBody(
+                Json.toJson(messageResponse).toString()
+              )
+          )
+      )
+
+      implicit val hc = HeaderCarrier()
+      val result      = persistenceConnector.getArrivalsForEori(eori)
+      whenReady(result) {
+        _ mustBe messageResponse
+      }
+    }
+
+    "on incorrect Json, return an error" in {
```

Comment:
I am confused with this test case. is this valid scenario to test ? @drnaylor-oc

Reply:
It's testing if an upstream failure occurs, but in a non-standard way. It's fine.

Reply:
@drnaylor-oc To add to that, I believe it's testing if the body returned can't be deserialised into the expected type